### PR TITLE
Load extensions from classpath, without scanning

### LINKED ISSUE #33: Use ServiceLoader for providing extensions
https://docs.oracle.com/javase/9/docs/api/java/util/ServiceLoader.html

Providing extensions like this: https://github.com/prestosql/presto/blob/master/presto-product-tests/src/main/java/io/prestosql/tests/TemptoProductTestRunner.java#L42
makes impossible to run tempto tests from IDE.

FYI: @findepi

### PR DESCRIPTION
**Pull request by findepi**
Fixes https://github.com/prestosql/tempto/issues/33

### CODE REVIEW

File: tempto-core/src/test/groovy/io/prestosql/tempto/internal/initialization/TestInitializationListenerTest.groovy [hunk 1]
```diff
@@ -20,6 +20,7 @@ import io.prestosql.tempto.context.State
 import io.prestosql.tempto.context.TestContext
 import io.prestosql.tempto.context.TestContextCloseCallback
 import io.prestosql.tempto.fulfillment.RequirementFulfiller
+import io.prestosql.tempto.fulfillment.RequirementFulfiller.TestLevelFulfiller
 import io.prestosql.tempto.fulfillment.TestStatus
 import io.prestosql.tempto.internal.TestSpecificRequirementsResolver
 import org.testng.*
```
Comment (kokosing) [inline tempto-core/src/test/groovy/io/prestosql/tempto/internal/initialization/TestInitializationListenerTest.groovy:26]: can you please remove `*`?

File: tempto-core/src/main/java/io/prestosql/tempto/fulfillment/table/TableDefinitionsRepository.java [hunk 2]
```diff
@@ -108,12 +67,13 @@ public static TableDefinition tableDefinition(TableHandle tableHandle)
     public static TableDefinitionsRepository tableDefinitionsRepository()
     {
         if (TABLE_DEFINITIONS_REPOSITORY == null) {
-            TABLE_DEFINITIONS_REPOSITORY = new TableDefinitionsRepository(ImmutableList.<TableDefinition>builder()
-                    .addAll(BUILTIN_TABLE_DEFINITIONS)
-                    .addAll(ADDITIONAL_TABLE_DEFINITIONS.get())
-                    // TODO: since TestNG has no listener that can be run before tests factory, this has to be initialized here
-                    .addAll(getAllConventionBasedTableDefinitions())
-                    .build());
+            List<TemptoPlugin> plugins = ImmutableList.copyOf(ServiceLoader.load(TemptoPlugin.class).iterator());
```
Comment (kokosing) [inline tempto-core/src/main/java/io/prestosql/tempto/fulfillment/table/TableDefinitionsRepository.java:70]: I would write some dedicated method to load plugins. `TemptoPlugins.load()`?

**Review by kokosing — APPROVED** [review]
Thanks for doing this!

File: tempto-core/src/test/groovy/io/prestosql/tempto/internal/initialization/TestInitializationListenerTest.groovy (hunk 1 above)
Comment (findepi) [reply tempto-core/src/test/groovy/io/prestosql/tempto/internal/initialization/TestInitializationListenerTest.groovy:26]: preexisting; intellij autoformat doesn't "fix" this.
ideally i'd rather see all the groovy gone

File: tempto-core/src/main/java/io/prestosql/tempto/fulfillment/table/TableDefinitionsRepository.java (hunk 2 above)
Comment (findepi) [reply tempto-core/src/main/java/io/prestosql/tempto/fulfillment/table/TableDefinitionsRepository.java:70]: Ideally the plugins should be loaded once, but I don't see a way to properly keep state in tempto suite.
for now, i'm leaving this one-liner as-is.

**Pull request merged**